joblib/test/test_parallel.py

```diff
@@ -664,24 +746,59 @@ def test_no_blas_crash_or_freeze_with_multiprocessing():
 
     # check that the internal BLAS thread-pool is not in an inconsistent state
     # in the worker processes managed by multiprocessing
-    Parallel(n_jobs=2, backend=spawn_backend)(
+    Parallel(n_jobs=2, backend=backend)(
         delayed(np.dot)(a, a.T) for i in range(2))
 
 
-def test_parallel_with_interactively_defined_functions():
-    # When functions are defined interactively in a python/IPython
-    # session, we want to be able to use them with joblib.Parallel
-    if posix is None:
-        # This test pass only when fork is the process start method
-        raise SkipTest('Not a POSIX platform')
-
-    code = '\n\n'.join([
+@with_multiprocessing
+@parametrize('backend', PROCESS_BACKENDS +
+             ([] if sys.version_info[:2] < (3, 4) or mp is None
+              else ['spawn']))
+def test_parallel_with_interactively_defined_functions(backend):
+    code = '\n'.join([
         'from joblib import Parallel, delayed',
-        'def square(x): return x**2',
-        'print(Parallel(n_jobs=2)(delayed(square)(i) for i in range(5)))'])
+        'if __name__ == "__main__":',
+        '    def square(x): return x**2',
+        '    backend="{}"'.format(backend),
+        '    if backend == "spawn":',
+        '        from multiprocessing import get_context',
+        '        backend = get_context(backend)',
+        '    print(Parallel(n_jobs=2, backend=backend)(',
+        '          delayed(square)(i) for i in range(5)))'])
+
+    fid, filename = mkstemp(suffix="_joblib.py")
+    os.close(fid)
+    try:
+        with open(filename, mode='wb') as f:
+            f.write(code.encode('ascii'))
+        check_subprocess_call([sys.executable, filename],
+                              stdout_regex=r'\[0, 1, 4, 9, 16\]',
+                              timeout=2)
+    finally:
+        os.unlink(filename)
 
-    check_subprocess_call([sys.executable, '-c', code],
-                          stdout_regex=r'\[0, 1, 4, 9, 16\]')
+
+@with_multiprocessing
+def test_parallel_with_interactively_defined_functions_default_backend():
+    # The default backend does not require if __name__ == '__main__'
+    code = """if True:
```

Comment:
The `if True` is just a trick to be able to arbitrarily indent the code in the string. The goal of this test is to actually show that we don't need `if __name__ == '__main__'`.